static_src/components/route_form.jsx

```diff
@@ -0,0 +1,129 @@
+
+import style from 'cloudgov-style/css/cloudgov-style.css';
+import React from 'react';
+
+import PanelAction from './panel_action.jsx';
+
+import createStyler from '../util/create_styler';
+
+import panelCss from '../css/panel.css';
+import routeFormCss from '../css/route_form.css';
+
+const propTypes = {
+  domains: React.PropTypes.array.isRequired,
+  route: React.PropTypes.object,
+  handleSubmit: React.PropTypes.func,
+  handleCancel: React.PropTypes.func
+};
+
+const defaultProps = {
+  route: {},
+  handleSubmit: () => {},
+  handleCancel: () => {}
+};
+
+export default class RouteForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      domain: props.route.domain,
+      host: props.route.host,
+      path: props.route.path
+    };
+    this._onChange = this._onChange.bind(this);
+    this.styler = createStyler(style, panelCss, routeFormCss);
+  }
+
+
+  // TODO: If there are multiple route forms on the page, does it matter if
+  // they all have elements with the same names? IDs are unique
+  _onChange(ev) {
+    const newValue = {};
+    newValue[ev.target.name] = ev.target.value;
+    this.setState(newValue);
+  }
+
+  get fullUrl() {
+    let url = `${this.state.host}.${this.state.domain}`;
+    if (this.state.path !== '') url += `/${this.state.path}`;
+    return url;
+  }
+
+  get hasChanged() {
+    let changed = false;
+
+    if (this.state.host !== this.props.route.host) {
+      changed = true;
+    } else if (this.state.domain !== this.props.route.domain) {
+      changed = true;
+    } else if (this.state.path !== this.props.route.path) {
+      changed = true;
+    }
+
+    return changed;
+  }
+
+  render() {
+    const route = this.props.route;
+    const domains = this.props.domains;
+
+    return (
+      <form className={ this.styler('route-form') }>
```

Comment:
Right now it definitely does. For example, the big "." and "/" in the UI are with CSS and pseudo elements. If we decide we want to make it into a generic component that could just be text in a component, but even then it would need some custom CSS.